Symfony/CS/Fixer/Contrib/ClassKeywordRemoveFixer.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class ClassKeywordRemoveFixer extends AbstractFixer
+{
+    /**
+     * @var string[]
+     */
+    private $imports = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        $this->storeImports($tokens);
+        $tokens->rewind();
+        $this->replaceClassKeywords($tokens);
+
+        return $tokens->generateCode();
+    }
+
+    private function storeImports(Tokens $tokens)
+    {
+        foreach ($tokens->getImportUseIndexes() as $index) {
+            $import = '';
+            while (($index = $tokens->getNextMeaningfulToken($index))) {
+                if ($tokens[$index]->equals(';')) {
+                    break;
+                }
+                $import .= $tokens[$index]->getContent();
+            }
+            $this->imports[] = $import;
+        }
+    }
+
+    private function replaceClassKeywords(Tokens $tokens)
+    {
+        foreach ($tokens->findGivenKind(T_DOUBLE_COLON) as $index => $token) {
+            if ('class' !== $tokens[$tokens->getNextMeaningfulToken($index)]->getContent()) {
+                continue;
+            }
+
+            $this->replaceClassKeyword($tokens, $index);
+        }
+    }
+
+    private function replaceClassKeyword(Tokens $tokens, $doubleColonIndex)
+    {
+        $classEndIndex = $doubleColonIndex - 1;
+        $classBeginIndex = $classEndIndex;
+        while ($tokens[--$classBeginIndex]->isGivenKind([T_NS_SEPARATOR, T_STRING]));
+        $classBeginIndex++;
+        $classString = $tokens->generatePartialCode($classBeginIndex, $classEndIndex);
+        $classImport = current(array_filter($this->imports, function ($import) use ($classString) {
+            $namespaceToTest = explode('\\', $classString)[0];
+
+            return 0 === strcmp($namespaceToTest, substr($import, - strlen($namespaceToTest)));
+        }));
+
+        $tokens->clearRange($classBeginIndex, $tokens->getNextMeaningfulToken($doubleColonIndex));
+        $tokens->insertAt($classBeginIndex, [
+            new Token("'"),
+            new Token($this->makeClassFQN($classImport, $classString)),
```

Comment:
Also with:

```php
$tokens->insertAt($classBeginIndex, new Token([T_STRING, "'".$this->makeClassFQN($classImport, $classString)."'"]));
```

Got:

```bash
1) Symfony\CS\Tests\Fixer\Contrib\ClassKeywordRemoveFixerTest::testFix with data set #0 ('<?php\n                use Foo...      ', '<?php\n                use Foo...      ')
The token at index 13 should be [318,"'Foo\\Bar\\Thing'"], got {
    "id": 310,
    "name": "T_STRING",
    "content": "'Foo\\Bar\\Thing'",
    "line": null,
    "isArray": true
}
Failed asserting that false is true.
```

The `318` id corresponds to a `T_DO` token, this make no sense...